src/dataframe/dataframe.jl

```diff
@@ -596,17 +596,22 @@ Base.setindex!(df::DataFrame, x::Void, col_ind::Int) = delete!(df, col_ind)
 
 Base.empty!(df::DataFrame) = (empty!(df.columns); empty!(index(df)); df)
 
-function Base.insert!(df::DataFrame, col_ind::Int, item::AbstractVector, name::Symbol)
+function Base.insert!(df::DataFrame, col_ind::Int, item::AbstractVector,
+                      name::Symbol; allow_duplicates=false)
     0 < col_ind <= ncol(df) + 1 || throw(BoundsError())
     size(df, 1) == length(item) || size(df, 1) == 0 || error("number of rows does not match")
 
+    if !allow_duplicates
+        name in _names && throw(ArgumentError("Duplicate column name $name"))
+    end
+
     insert!(index(df), col_ind, name)
     insert!(df.columns, col_ind, item)
     df
 end
 
-function Base.insert!(df::DataFrame, col_ind::Int, item, name::Symbol)
-    insert!(df, col_ind, upgrade_scalar(df, item), name)
+function Base.insert!(df::DataFrame, col_ind::Int, item, name::Symbol; allow_duplicates=true)
```

Comment:
`allow_duplicates=true` -> `allow_duplicates=false`?

Reply:
of course - I have changed my mind to propose a breaking change and forgotten to update it here. Fixed.